python/instrumentation/openinference-instrumentation-haystack/src/openinference/instrumentation/haystack/_wrappers.py

```diff
@@ -0,0 +1,254 @@
+
+                attributes.update(
+                    _flatten(
+                        {
+                            f"{emb_vec_0}{EMBEDDING_VECTOR}": f"<{emb_len} dimensional vector>",
+                            f"{emb_vec_0}{EMBEDDING_TEXT}": invocation_parameters["text"],
+                        }
+                    )
+                )
+            elif component_type == "retriever":
+                attributes.update(
+                    _flatten(
+                        {
+                            SpanAttributes.OUTPUT_VALUE: safe_json_dumps(response["documents"]),
+                            SpanAttributes.OUTPUT_MIME_TYPE: OpenInferenceMimeTypeValues.JSON,
+                        }
+                    )
+                )
+
+                for i, document in enumerate(response["documents"]):
+                    attributes[
+                        f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_CONTENT}"
+                    ] = document.content
+                    attributes[
+                        f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_ID}"
+                    ] = document.id
+                    attributes[
+                        f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_SCORE}"
+                    ] = document.score
+                    attributes[
+                        f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_METADATA}"
+                    ] = safe_json_dumps(document.meta)
+
+            span.set_attributes(attributes)
+
+        return response
+
+
+class _PipelineWrapper(_WithTracer):
+    """
+    Wrapper for the pipeline processing
+    Captures all calls to the pipeline
+    """
+
+    def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        if context_api.get_value(context_api._SUPPRESS_INSTRUMENTATION_KEY):
+            return wrapped(*args, **kwargs)
+
+        # Prepare invocation parameters by merging args and kwargs
+        invocation_parameters = {}
+        for arg in args:
+            if arg and isinstance(arg, dict):
+                invocation_parameters.update(arg)
+        invocation_parameters.update(kwargs)
+
+        span_name = "Pipeline"
+        with self._tracer.start_as_current_span(
+            span_name,
+            attributes=dict(
+                _flatten(
+                    {
+                        SpanAttributes.OPENINFERENCE_SPAN_KIND: CHAIN,
+                        SpanAttributes.INPUT_VALUE: safe_json_dumps(invocation_parameters),
+                        SpanAttributes.INPUT_MIME_TYPE: OpenInferenceMimeTypeValues.JSON,
+                    }
+                )
+            ),
+        ) as span:
+            span.set_attributes(dict(get_attributes_from_context()))
```

Comment:
```suggestion
            attributes={
                **_flatten(
                    {
                        SpanAttributes.OPENINFERENCE_SPAN_KIND: CHAIN,
                        SpanAttributes.INPUT_VALUE: safe_json_dumps(invocation_parameters),
                        SpanAttributes.INPUT_MIME_TYPE: OpenInferenceMimeTypeValues.JSON,
                    }
                ),
                **get_attributes_from_context()
            },
        ) as span:
```

Non-blocking suggestion.